numba/targets/arraymath.py

```diff
@@ -2640,3 +2640,29 @@ def impl(a, dtype=None):
                 return np.array(a, ty)
 
     return impl
+
+@overload(np.extract)
+def np_extract(condition, arr):
+
+    def np_extract_impl(condition, arr):
+        cond = np.asarray(condition)
+        a = np.asarray(arr)
+
+        if a.size == 0:
+            raise ValueError('Cannot extract from an empty array')
+
+        # the below looks odd but replicates NumPy
+        msg = 'condition shape inconsistent with arr shape'
+        if a.ndim == 0 and cond.size > 1:
+            raise ValueError(msg)
+
+        if a.ndim == 1 and cond.size > a.size + 1:
```

Comment:
hmmm this is strange... however NumPy does IndexError for:
```
np.extract(np.array([True, False, False, False, True]), np.arange(4))
```
whereas this impl returns `array[0]` ?

Reply:
Added test